### PR TITLE
MAPREDUCE-7240.Exception 'Invalid event: TA_TOO_MANY_FETCH_FAILURE at…

### PR DESCRIPTION
**Pull request by chimney-lee**
… SUCCESS_FINISHING_CONTAINER' cause job error

https://github.com/apache/hadoop/pull/1618
PR for trunk

### CODE REVIEW

File: hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app/src/test/java/org/apache/hadoop/mapreduce/v2/app/job/impl/TestTaskAttempt.java [hunk 1]
```diff
@@ -1870,6 +1870,31 @@ private void initResourceTypes() {
...
+    TaskId reducetaskId = MRBuilderUtils.newTaskId(taImpl.getID().getTaskId()
+        .getJobId(), 1, TaskType.REDUCE);
+    TaskAttemptId reduceTAId =
+        MRBuilderUtils.newTaskAttemptId(reducetaskId, 0);
+
+    taImpl.handle(new TaskAttemptEvent(taImpl.getID(),
+        TaskAttemptEventType.TA_DONE));
+
+    assertEquals("Task attempt's internal state is not " +
+        "SUCCESS_FINISHING_CONTAINER",
```
Comment (hadoop-yetus) [inline hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app/src/test/java/org/apache/hadoop/mapreduce/v2/app/job/impl/TestTaskAttempt.java]: whitespace:end of line

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 71 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1250 | trunk passed |
| +1 | compile | 29 | trunk passed |
| +1 | checkstyle | 27 | trunk passed |
| +1 | mvnsite | 31 | trunk passed |
| +1 | shadedclient | 887 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 25 | trunk passed |
| 0 | spotbugs | 64 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 62 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 35 | the patch passed |
| +1 | compile | 28 | the patch passed |
| +1 | javac | 28 | the patch passed |
| -0 | checkstyle | 24 | hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app: The patch generated 1 new + 247 unchanged - 0 fixed = 248 total (was 247) |
| +1 | mvnsite | 30 | the patch passed |
| -1 | whitespace | 0 | The patch has 1 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply |
| +1 | shadedclient | 1013 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 27 | the patch passed |
| +1 | findbugs | 66 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 622 | hadoop-mapreduce-client-app in the patch passed. |
| +1 | asflicense | 38 | The patch does not generate ASF License warnings. |
| | | 4366 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.4 Server=19.03.4 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1674/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1674 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 9fa34c3cb237 4.15.0-66-generic #75-Ubuntu SMP Tue Oct 1 05:24:09 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / ac6b6a6 |
| Default Java | 1.8.0_222 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1674/1/artifact/out/diff-checkstyle-hadoop-mapreduce-project_hadoop-mapreduce-client_hadoop-mapreduce-client-app.txt |
| whitespace | https://builds.apache.org/job/hadoop-multibranch/job/PR-1674/1/artifact/out/whitespace-eol.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1674/1/testReport/ |
| Max. process+thread count | 584 (vs. ulimit of 5500) |
| modules | C: hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app U: hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1674/1/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 2538 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1393 | trunk passed |
| +1 | compile | 29 | trunk passed |
| +1 | checkstyle | 26 | trunk passed |
| +1 | mvnsite | 32 | trunk passed |
| +1 | shadedclient | 885 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 24 | trunk passed |
| 0 | spotbugs | 57 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 54 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 31 | the patch passed |
| +1 | compile | 25 | the patch passed |
| +1 | javac | 25 | the patch passed |
| -0 | checkstyle | 21 | hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app: The patch generated 1 new + 247 unchanged - 0 fixed = 248 total (was 247) |
| +1 | mvnsite | 27 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 907 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 20 | the patch passed |
| +1 | findbugs | 58 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 606 | hadoop-mapreduce-client-app in the patch passed. |
| +1 | asflicense | 28 | The patch does not generate ASF License warnings. |
| | | 6808 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.4 Server=19.03.4 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1674/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1674 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 664042e4db45 4.15.0-66-generic #75-Ubuntu SMP Tue Oct 1 05:24:09 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 30ed24a |
| Default Java | 1.8.0_222 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1674/2/artifact/out/diff-checkstyle-hadoop-mapreduce-project_hadoop-mapreduce-client_hadoop-mapreduce-client-app.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1674/2/testReport/ |
| Max. process+thread count | 581 (vs. ulimit of 5500) |
| modules | C: hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app U: hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1674/2/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 1881 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1182 | trunk passed |
| +1 | compile | 30 | trunk passed |
| +1 | checkstyle | 33 | trunk passed |
| +1 | mvnsite | 38 | trunk passed |
| +1 | shadedclient | 816 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 25 | trunk passed |
| 0 | spotbugs | 56 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 54 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 31 | the patch passed |
| +1 | compile | 27 | the patch passed |
| +1 | javac | 27 | the patch passed |
| -0 | checkstyle | 23 | hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app: The patch generated 1 new + 247 unchanged - 0 fixed = 248 total (was 247) |
| +1 | mvnsite | 29 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 823 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 23 | the patch passed |
| +1 | findbugs | 63 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 627 | hadoop-mapreduce-client-app in the patch passed. |
| +1 | asflicense | 33 | The patch does not generate ASF License warnings. |
| | | 5839 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.4 Server=19.03.4 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1674/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1674 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux d32faf520c94 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 30ed24a |
| Default Java | 1.8.0_222 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1674/3/artifact/out/diff-checkstyle-hadoop-mapreduce-project_hadoop-mapreduce-client_hadoop-mapreduce-client-app.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1674/3/testReport/ |
| Max. process+thread count | 727 (vs. ulimit of 5500) |
| modules | C: hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app U: hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1674/3/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by chimney-lee** [comment]
@aajisaka thanx if have a review for this!

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m  0s |  Docker mode activated.  |
| -1 :x: |  patch  |   0m  4s |  https://github.com/apache/hadoop/pull/1674 does not apply to trunk. Rebase required? Wrong Branch? See https://wiki.apache.org/hadoop/HowToContribute for help.  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hadoop/pull/1674 |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-1674/3/console |
| versions | git=2.17.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m  0s |  Docker mode activated.  |
| -1 :x: |  patch  |   0m 10s |  https://github.com/apache/hadoop/pull/1674 does not apply to trunk. Rebase required? Wrong Branch? See https://wiki.apache.org/hadoop/HowToContribute for help.  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hadoop/pull/1674 |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-1674/4/console |
| versions | git=2.17.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request closed without merging**